### PR TITLE
protonmail-bridge: 1.2.3-1 -> 1.2.6-1

### PR DESCRIPTION
**Pull request by CloudArtAdmins**
###### Motivation for this change
Wanted more up to date version of package

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by apeyroux — CHANGES_REQUESTED** [review]
1. the file ProtonMail_Bridge.desktop is no longer named ProtonMail_Bridge.desktop but protonmail-bridge.desktop
2. this new version needs Qt_5.13 and for the moment it is not available in nixpkgs  (https://github.com/NixOS/nixpkgs/issues/74355).
3. we do not find the same sha256 for the src
```patch
@@ -2,7 +2,7 @@
 , libsecret, libGL, libpulseaudio, glib, wrapQtAppsHook, mkDerivation }:
 
 let
-  version = "1.2.3-1";
+  version = "1.2.6-1";
 
   description = ''
     An application that runs on your computer in the background and seamlessly encrypts
@@ -16,7 +16,7 @@ in mkDerivation {
 
   src = fetchurl {
     url = "https://protonmail.com/download/protonmail-bridge_${version}_amd64.deb";
-    sha256 = "032ggk9fvd19fbsqkzwzwh0hpyg8gpkrin71di7zsx6ias5innw1";
+    sha256 = "0iy2x9qnvhqrv6df0mjmxss9jddd64gh8wbadkn0cvrczcbwkz9l";
   };
 
   sourceRoot = ".";
@@ -55,7 +55,7 @@ in mkDerivation {
       --set-rpath "${rpath}" \
       $out/lib/protonmail-bridge
 
-    substituteInPlace $out/share/applications/ProtonMail_Bridge.desktop \
+    substituteInPlace $out/share/applications/protonmail-bridge.desktop \
       --replace "/usr/" "$out/" \
       --replace "Exec=protonmail-bridge" "Exec=$out/bin/protonmail-bridge"
   '';

```

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 1]
```diff
@@ -55,7 +55,7 @@ in mkDerivation {
       --set-rpath "${rpath}" \
       $out/lib/protonmail-bridge
 
-    substituteInPlace $out/share/applications/ProtonMail_Bridge.desktop \
+    substituteInPlace $out/share/applications/protonmail_bridge.desktop \
```
Comment (apeyroux) [inline pkgs/applications/networking/protonmail-bridge/default.nix:58]: `s/protonmail_bridge/protonmail-bridge/`

**Review by apeyroux — CHANGES_REQUESTED** [review]
`s/protonmail_bridge/protonmail-bridge/`

File: pkgs/applications/networking/protonmail-bridge/default.nix (hunk 1 above)
Comment (lightdiscord) [inline pkgs/applications/networking/protonmail-bridge/default.nix:58]: I can't build the package, there is no `protonmail_bridge.desktop` file, but there is a `protonmail-bridge.desktop` file.

Comment (privateseabass) [reply pkgs/applications/networking/protonmail-bridge/default.nix:58]: Oh! Thank you.

**Comment by privateseabass** [comment]
Now awaiting an update for the qt package to update.
(#74355 )

**Comment by apeyroux** [comment]
We'll probably have to group the commit messages into one.

**Comment by privateseabass** [comment]
Fine by me.

**Comment by apeyroux** [comment]
I tested the build with this pr:https://github.com/NixOS/nixpkgs/pull/84232 and 

```diff
modified   pkgs/top-level/all-packages.nix
@@ -21348,7 +21348,7 @@ in
 
   properties-cpp = callPackage ../development/libraries/properties-cpp { };
 
-  protonmail-bridge = libsForQt512.callPackage ../applications/networking/protonmail-bridge { };
+  protonmail-bridge = libsForQt514.callPackage ../applications/networking/protonmail-bridge { };
 
   protonvpn-cli = callPackage ../applications/networking/protonvpn-cli { };
 ```

1.2.6-1 build without errors and cli works but by using X there is this message: 

> ./result/bin/protonmail-bridge: symbol lookup error: ./result/bin/protonmail-bridge: undefined symbol: _ZdlPvm, version Qt_5

Maybe a QT specialist has an idea? :)

**Comment by GRBurst** [comment]
This error has been posted at [stackoverflow](https://stackoverflow.com/a/53234995)

**Comment by privateseabass** [comment]
Protonmail has open-sourced the linux app for protonmail bridge.
So an alternative might be to create an app for cli only or maybe the code as clearer information on the issue.

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 2]
```diff
@@ -16,7 +16,7 @@ in mkDerivation {
 
   src = fetchurl {
     url = "https://protonmail.com/download/protonmail-bridge_${version}_amd64.deb";
-    sha256 = "032ggk9fvd19fbsqkzwzwh0hpyg8gpkrin71di7zsx6ias5innw1";
+    sha256 = "34fdc917fb2c6f06ec6c6a71041f31ad3599b4ee5556e09ad919c36d71eac247";
```
Comment (gamb) [inline pkgs/applications/networking/protonmail-bridge/default.nix:19]: Is this correct? I got `0iy2x9qnvhqrv6df0mjmxss9jddd64gh8wbadkn0cvrczcbwkz9l`

Comment (privateseabass) [reply pkgs/applications/networking/protonmail-bridge/default.nix:19]: The one I input used a sha256, not the nix-prefix.
Might be worth updating the version as well, as I think there has been a later update, by this point.

Comment (gamb) [reply pkgs/applications/networking/protonmail-bridge/default.nix:19]: Cheers @privateseabass hadn't realized that was an option.

Comment (privateseabass) [reply pkgs/applications/networking/protonmail-bridge/default.nix:19]: No worries.
Most people hadn't.

**Comment by IvarWithoutBones** [comment]
Why was this closed?

**Comment by privateseabass** [comment]
A newer version was released and.. Really don't know if there is a more proper process for an outdated branch.

**Comment by talon** [comment]
@IvarWithoutBones I believe this is closed because 1) there's a newer version out but more importantly 2) [ever since 1.2.4 the bridge depends on Qt 5.13](https://github.com/ProtonMail/proton-bridge/blob/master/Changelog.md#v124-brooklyn-beta-2019-12-16) which means we have to wait for https://github.com/NixOS/nixpkgs/pull/84232 to be merged before protonmail-bridge can be updated to the latest version again

**Pull request closed without merging**